ci: Only log into Conan when uploading packages

bthomee opened the pull request:
## High Level Overview of Change

The `conan remote login` step only needs to be performed when uploading packages.

### Context of Change

There are separate steps for logging into Conan and uploading packages. However, at the moment sometimes the login step is executed even though no packages will be uploaded. The condition for performing both steps should be the same.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [X] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

mathbunnyru commented on .github/workflows/upload-conan-deps.yml:
```diff
@@ -84,7 +84,7 @@ jobs:
 
       - name: Log into Conan remote
-        if: ${{ github.repository_owner == 'XRPLF' && github.event_name != 'pull_request' }}
+        if: ${{ github.repository_owner == 'XRPLF' && github.event_name != 'pull_request' && github.event_name != 'schedule' }}
```
```suggestion
        if: ${{ github.repository_owner == 'XRPLF' && (github.event_name == 'workflow_dispatch' || github.event_name == 'push') }}
```

I think at this point it's more clear to tell when we do login (and maybe upload) (and change the condition below)

bthomee replied:
Done. Applied both changes in the second commit.

bthomee committed:
ci: Only log into Conan when uploading packages

There are separate steps for logging into Conan and uploading packages. However, at the moment sometimes the login step is executed even though no packages will be uploaded. The condition for performing both steps should be the same.
